plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/DefaultValidationIssueStore.java

```diff
@@ -0,0 +1,207 @@
+ * Contributors:
+ * 	committers of YAKINDU - initial API and implementation
+ * 
+ */
+package org.yakindu.sct.ui.editor.validation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.workspace.util.WorkspaceSynchronizer;
+import org.eclipse.gmf.runtime.common.ui.resources.FileChangeManager;
+import org.eclipse.gmf.runtime.common.ui.resources.IFileObserver;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
+import org.eclipse.xtext.ui.util.IssueUtil;
+import org.eclipse.xtext.validation.Issue;
+import org.yakindu.sct.model.sgraph.ui.validation.SCTIssue;
+import org.yakindu.sct.ui.editor.DiagramActivator;
+import org.yakindu.sct.ui.editor.preferences.StatechartPreferenceConstants;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import com.google.inject.Inject;
+
+/**
+ * 
+ *
+ * @author Andreas Mülder - Initial contribution and API
+ *
+ */
+public class DefaultValidationIssueStore
+		implements IValidationIssueStore, IFileObserver, IMarkerType, IPropertyChangeListener {
+
+	@Inject
+	private IssueUtil issueCreator;
+
+	private List<IResourceIssueStoreListener> listener;
+	// the URI of the notation element
+	private Multimap<String, SCTIssue> persistentIssues;
+	private Multimap<String, SCTIssue> liveIssues;
+	private boolean connected = false;
+
+	private Resource resource;
+
+	public DefaultValidationIssueStore() {
+		listener = Lists.newArrayList();
+		persistentIssues = ArrayListMultimap.create();
+		liveIssues = ArrayListMultimap.create();
+		DiagramActivator.getDefault().getPreferenceStore().addPropertyChangeListener(this);
+	}
+
+	protected String getMarkerType() {
+		return SCT_MARKER_TYPE;
+	}
+
+	@Override
+	public void addIssueStoreListener(IResourceIssueStoreListener listener) {
+		this.listener.add(listener);
+	}
+
+	@Override
+	public void removeIssueStoreListener(IResourceIssueStoreListener listener) {
+		this.listener.remove(listener);
+	}
+
+	protected void notifyListeners() {
+		for (IResourceIssueStoreListener iResourceIssueStoreListener : listener) {
+			iResourceIssueStoreListener.issuesChanged();
+		}
+	}
+
+	@Override
+	public void connect(Resource resource) {
+		if (connected)
+			throw new IllegalStateException("Issue store is already connected to a resource");
+		connected = true;
+		this.resource = resource;
+		reloadMarkerIssues();
+	}
+
+	protected void reloadMarkerIssues() {
+		persistentIssues.clear();
+		if (liveValidationEnabled())
+			return;
+		IFile file = WorkspaceSynchronizer.getFile(resource);
```

Comment:
In SCT there is the option to disable live validation in the preferences. If you do that, you still want to see the persistent error markers in your editor. Thus, the connection to the marker store.

This is also preliminary work for stale error markers. These are error markers that are present as a marker (unsafed changes) but already fixed in the editor. In JDT and Xtext editors, they are visualized as grayscale error icons.  To calculate stale error markers I need the IMarker based issues aswell.